### PR TITLE
expose maskData on GroupLayer

### PR DESCRIPTION
**Pull request by alexspevak**

### CODE REVIEW

File: packages/psd/src/sections/LayerAndMaskInformation/index.ts [hunk 1]
```diff
@@ -105,24 +105,25 @@ export function parseLayerAndMaskInformation(
...
-
       const frame: Frame | undefined = stack.pop();
       if (frame === undefined) {
         throw new PanicFrameStackUnmatched();
       }
 
       const groupId = frame.groupId > 0 ? frame.groupId : undefined;
       const _layerRecord = frame.layerRecord || layerRecord;
-
+      const _channels = frame.channels || channels;
```
Comment (scoiatael) [inline packages/psd/src/sections/LayerAndMaskInformation/index.ts:120]: Why this alternative? For `LayerFrame` we use `channels` directly. I don't think `frame` contains `channels` property anyway.

File: packages/psd/src/sections/LayerAndMaskInformation/index.ts [hunk 2]
```diff
@@ -105,24 +105,25 @@ export function parseLayerAndMaskInformation(
         groupId: alreadyViewed,
         parentGroupId: currentGroupId,
         layerRecord,
+        channels,
       });
       orders.push("G");
     } else if (dividerType === GroupDivider.BoundingSection) {
       // Indicates the end of a layer group
-
```
Comment (scoiatael) [inline packages/psd/src/sections/LayerAndMaskInformation/index.ts:112]: Why remove empty line?

File: packages/psd/src/classes/Group.ts [hunk 3]
```diff
@@ -68,4 +70,30 @@ export class Group implements NodeBase<NodeParent, NodeChild> {
   get id(): number | undefined {
     return this.layerFrame?.id;
   }
+
+  async userMask(): Promise<Uint8Array | undefined> {
+    const userMask = this.layerFrame?.userMask;
+
+    if (!userMask) {
+      return undefined;
```
Comment (scoiatael) [inline packages/psd/src/classes/Group.ts:78]: Would be nice to have some consistency here between `return` vs `return undefined` ;)

File: packages/psd/src/classes/Group.ts [hunk 4]
```diff
@@ -68,4 +70,30 @@ export class Group implements NodeBase<NodeParent, NodeChild> {
...
+    }
+    const {maskData} = this;
+
+    if (!maskData) {
```
Comment (scoiatael) [inline packages/psd/src/classes/Group.ts:82]: Why not combine both ifs like in `realUserMask`?

File: packages/psd/src/sections/LayerAndMaskInformation/index.ts (hunk 1 above)
Comment (alexspevak) [reply packages/psd/src/sections/LayerAndMaskInformation/index.ts:120]: are you saying that frame.channels should not be there?  in previous block we do stack.push({layerRecord}) (stack is Frame). we need corresponding channels, so I expanded Frame so it contains channels too

File: packages/psd/src/classes/Group.ts (hunk 4 above)
Comment (alexspevak) [reply packages/psd/src/classes/Group.ts:82]: done

File: packages/psd/src/classes/Group.ts (hunk 3 above)
Comment (alexspevak) [reply packages/psd/src/classes/Group.ts:78]: removed undefined

File: packages/psd/src/sections/LayerAndMaskInformation/index.ts (hunk 2 above)
Comment (alexspevak) [reply packages/psd/src/sections/LayerAndMaskInformation/index.ts:112]: returned

**Pull request closed without merging**